LoopKitUI/View Controllers/GlucoseRangeScheduleTableViewController.swift

```diff
@@ -168,147 +350,189 @@ public class GlucoseRangeScheduleTableViewController: DailyValueScheduleTableVie
         }
     }
 
     public override func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
-        switch Section(rawValue: section)! {
-        case .schedule:
-            return nil
+        switch sections[section] {
         case .override:
             return LocalizedString("Overrides", comment: "The section title of glucose overrides")
+        default:
+            return nil
+        }
+    }
+
+    public override func tableView(_ tableView: UITableView, titleForFooterInSection section: Int) -> String? {
+        switch sections[section] {
+        case .schedule:
+            return LocalizedString("Correction range is the blood glucose range that you would like Loop to correct to.", comment: "The section footer of correction range schedule")
+        default:
+            return nil
         }
     }
 
     // MARK: - UITableViewDelegate
 
     public override func tableView(_ tableView: UITableView, shouldHighlightRowAt indexPath: IndexPath) -> Bool {
-        switch Section(rawValue: indexPath.section)! {
-        case .schedule:
-            return super.tableView(tableView, shouldHighlightRowAt: indexPath)
-        case .override:
-            return false
-        }
+        return true
     }
 
     public override func tableView(_ tableView: UITableView, willSelectRowAt indexPath: IndexPath) -> IndexPath? {
-        switch Section(rawValue: indexPath.section)! {
+        tableView.beginUpdates()
+        switch sections[indexPath.section] {
         case .schedule:
-            return super.tableView(tableView, willSelectRowAt: indexPath)
+            updateTimeLimits(for: indexPath.row)
+            hideGlucoseRangeCells(excluding: indexPath)
         case .override:
-            return nil
+            hideGlucoseRangeCells(excluding: indexPath)
+        default:
+            break
         }
+        tableView.endEditing(false)
+
+        return indexPath
+    }
+
+    public override func tableView(_ tableView: UITableView, didDeselectRowAt indexPath: IndexPath) {
+        tableView.beginUpdates()
+        tableView.endUpdates()
     }
 
     public override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
-        switch Section(rawValue: indexPath.section)! {
-        case .schedule:
-            super.tableView(tableView, didSelectRowAt: indexPath)
-        case .override:
+        switch sections[indexPath.section] {
+        case .save:
+            delegate?.saveSchedule(for: self, completion: { (result) in
+                switch result {
+                case .success:
+                    self.isScheduleModified = false
+                    self.updateInsertButton()
+                case .failure(let error):
+                    self.present(UIAlertController(with: error), animated: true)
+                }
+            })
+        default:
             break
         }
+        tableView.endEditing(false)
+        tableView.endUpdates()
```

Comment:
I don't see a matching `beginUpdates()`.

Reply:
It's in willSelectRow